TouchPortalAPI/sdk_tools.py

```diff
@@ -551,7 +573,9 @@ def main(sdk_args=None):
             opts.target = output_path
 
     if opts.validate:
-        if entry_str:
+        if opts.target.endswith(".py"): # checks if is python file if It is then It will vaildate the python file by converting it to json first
+            valid = _validateDefinition(generateDefinitionFromScript(opts.target), as_str=True) # little hacky lol
+        elif entry_str:
```

Comment:
`if entry_str:` needs to come first, otherwise the .py file they're _generating from_ will be converted _again_ and then validated.  `entry_str` is only filled if the generator has run first.

Reply:
Sorry. I didn't noticed that. Should be fixed now. Ty